src/general/GameEventDescription.cs

```diff
@@ -0,0 +1,9 @@
+﻿/// <summary>
+///   A text-based description of what has happened in a game environment. Decorated with an icon if there's any.
+/// </summary>
+public class GameEventDescription
+{
+    public LocalizedString Description;
+    public string IconPath;
+    public bool Highlighted;
```

Comment:
I think these fields should all be documented, also changing these all to properties would be nice (maybe with private sets, so that no one edits a game event after it has been created).

Reply:
`readonly` modifier is maybe good for this.

Reply:
I kinda think that properties (get only properties would work here) make more sense from an OOP perspective. I'm still not really clear on best C# practices but it seems that public properties for a general class are really frowned upon. That's why I think this data class should use properties.

Reply:
Sure, but imo they're just kinda the same.

Reply:
I think this is a similar thing as the `struct` vs `class` distinction in C++, really the only thing different is the default access modifier of members, but selecting class versus struct actually conveys information to the reader as styleguides say to use them for different purposes. Struct for plain data and class for actual objects that hide their state in private variables and provide methods to access their functionality.